### PR TITLE
Fix/kfs 968 tackle zombie queries

### PR DESCRIPTION
**Pull request by YannickPferr**
Checklist
---------
1. [CRUCIAL] Is the change for features that are already live in prod?
   * yes: ok

What
----

This should fix the issue where some statements would not be deleted correctly. It also fixes future issues, where a request to the Gateway would be made with an invalid token. 

We now ensure that before making a request to the Gateway we at least tried to refresh the token.

### CODE REVIEW

File: internal/pkg/flink/internal/store/store.go [hunk 1]
```diff
@@ -243,12 +252,13 @@ func extractPageToken(nextUrl string) (string, error) {
...
 }
 
-func NewStore(client ccloudv2.GatewayClientInterface, exitApplication func(), appOptions *types.ApplicationOptions) StoreInterface {
+func NewStore(client ccloudv2.GatewayClientInterface, exitApplication func(), appOptions *types.ApplicationOptions, authenticated func() error) StoreInterface {
 	return &Store{
 		Properties:      appOptions.GetDefaultProperties(),
 		client:          client,
 		exitApplication: exitApplication,
 		appOptions:      appOptions,
+		authenticated:   authenticated,
```
Comment (swist) [inline internal/pkg/flink/internal/store/store.go:261]: this should be `tokenRefreshFunc`

File: internal/pkg/flink/internal/store/store.go [hunk 2]
```diff
@@ -32,6 +32,14 @@ type Store struct {
 	exitApplication func()
 	client          ccloudv2.GatewayClientInterface
 	appOptions      *types.ApplicationOptions
+	authenticated   func() error
+}
+
+func (s *Store) authenticatedGatewayClient() ccloudv2.GatewayClientInterface {
+	if authErr := s.authenticated(); authErr != nil {
+		log.CliLogger.Warnf("Failed to refresh token: %v", authErr)
```
Comment (swist) [inline internal/pkg/flink/internal/store/store.go:40]: I think this should error out. If we can't a good gateway client then we can't do anything?

**Review by swist — CHANGES_REQUESTED** [review]

File: internal/pkg/flink/internal/store/store.go (hunk 2 above)
Comment (YannickPferr) [reply internal/pkg/flink/internal/store/store.go:40]: Yes agree, but when this fails to refresh the token, the following request to the GW will error anyways -> and then we return  a proper error  to the user. I think that should be enough

File: internal/pkg/flink/internal/store/store.go (hunk 1 above)
Comment (YannickPferr) [reply internal/pkg/flink/internal/store/store.go]: Renamed!

File: internal/pkg/flink/internal/store/store.go (hunk 2 above)
Comment (swist) [reply internal/pkg/flink/internal/store/store.go:40]: sounds good to me

**Review by swist — APPROVED** [review]

**Pull request merged**